src/components/tree/tree.sandbox.ts

```diff
@@ -0,0 +1,344 @@
+import Vue, { PluginObject } from 'vue';
+import Component from 'vue-class-component';
+
+import { TREE_NAME } from '../component-names';
+import { TreeNode } from './tree';
+import WithRender from './tree.sandbox.html';
+
+export class Tree {
+    idNode: string;
+    elementLabel?: string;
+    uselessData?: string;
+    hasChildren?: boolean;
+}
+
+@WithRender
+@Component
+export class MRootTreeSandbox extends Vue {
+    public currentFile: string[] = ['/dossier 1/dossier 2/index.html'];
+    public currentFile2: string[] = ['/1/2'];
+    public wrongCurrentFile: string[] = ['/3/4'];
+
+    public emptyTree: TreeNode<Tree>[] = [];
+
+    public tree: TreeNode<Tree>[] = [
+        {
+            content: {
+                idNode: '1',
+                elementLabel: 'Titre 1',
+                uselessData: 'test',
+                hasChildren: true
```

Comment:
Why do we have to provide 'hasChildren' ? It would be easier to simply look at the "children" collection.

Reply:
Because empty folder (i.e. folder without any children) in a file tree must still behave like a parent in the tree, so that it we can open and close it. So we can't rely on the children list to decide wether we display it as a parent or as child.

Reply:
Ainsi que pour les répertoires (ou autre type de noeuds) dont le contenu est loadé à la demande seulement. Par contre, cette prop ne devrait pas avoir à être spécifiée à tout coup. Je n'ai pas regardé le code, mais comme le dit Fred, s'il y a déjà des children, hasChildren devrait être automatiquement calculé à true.

Reply:
Dans le fond quand je vérifie si c'est un dossier, je regarde d'abord s'il a des enfants et par la suite si il peut en avoir?
" !!node.children.length || node.hasChildren "